Refactored routing to utilize service architecture for endpoints.

CCondeluci opened the pull request:
+Added deck validation helper (currently disabled). Tested, included for review.

CCondeluci committed:
Refactored routing to utilize service architecture for endpoints.
+Added deck validation helper (currently disabled). Tested, included for review.

SpringsTea commented:
The refactor stuff looks good.

For the validator, I think we will have to specifically identify army cards. I don't want to rely on the presence and exact translation of the ability string. This also dosnt account for cards like endless8 and Miku that let you run up to 8 copies. I say pull the army logic out for now and we can make a mapping for these specific cards.

Also, check for <= 8 climaxes, less than 8 is still a legal deck, and there are conceptual advantages to this.

CCondeluci commented:
The "up to 8" and endless 8 stuff slipped my mind, I'll make those changes as soon as I can get some time, likely tomorrow evening.

CCondeluci committed:
Updated validator logic to accept <= 8 climaxes.
+ Updated validator logic to account for cards with art variants ("Mobility Gear", etc...)
+ Updated army logic to utilize mapping.
+ Added sample army-map.js mappings file that maps specific cards to their legal quantities.

~ Left deck validation disabled via comment, for later implementation when mappings are completed.

SpringsTea commented:
Great. I'll do some changes to the front end validation before enabling the deck validation stuff.
